packages/react-router/src/router.ts

```diff
@@ -456,19 +401,6 @@ export interface RouterOptions<
    * @link [Guide](https://tanstack.com/router/latest/docs/framework/react/guide/not-found-errors#default-router-wide-not-found-handling)
    */
   defaultNotFoundComponent?: NotFoundRouteComponent
-  /**
-   * The transformer that will be used when sending data between the server and the client during SSR.
-   *
-   * @link [API Docs](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType#transformer-property)
-   * @link [Guide](https://tanstack.com/router/latest/docs/framework/react/guide/ssr#data-transformers)
-   */
-  transformer?: RouterTransformer
```

Comment:
At this current stage, the removal of the user-configurable `transformer` property would be a breaking change. 

If the argument is that since this has the most impact on Start and that it's fine to make these breaking changes, could we perhaps move these Start-specific options under their own key?

Something like this perhaps?

```ts
createRouter({
	// ...
	server: { // or framework / start
		serializer: {}
	}
})
```

---

Reply:
I think we need a different API approach to configure server stuff. Ideally, it does not touch the router at all.